internal/testing/e2e/backfill_test.go

```diff
@@ -65,4 +65,37 @@ func TestCreateGetBackfill(t *testing.T) {
 	require.NotNil(t, err)
 	require.Equal(t, codes.NotFound.String(), status.Convert(err).Code().String())
 	require.Nil(t, actual)
+	require.Equal(t, b1, actual)
+}
+
+// TestBackfillFrontendLifecycle Update Backfill test
+func TestBackfillFrontendLifecycle(t *testing.T) {
+	om := newOM(t)
+	ctx := context.Background()
+
+	bf := &pb.Backfill{SearchFields: &pb.SearchFields{
+		StringArgs: map[string]string{
+			"search": "me",
+		},
+	},
+	}
+	createdBf, err := om.Frontend().CreateBackfill(ctx, &pb.CreateBackfillRequest{Backfill: bf})
+	require.Nil(t, err)
+
+	createdBf.SearchFields.StringArgs["key"] = "val"
+	updatedBf, err := om.Frontend().UpdateBackfill(ctx, &pb.UpdateBackfillRequest{Backfill: createdBf})
```

Comment:
To better test this method, send an entirely new Backfill protobuf object, with the new value but setting the ID to the one from createdBf.

As it is, this test incorrectly thinks it verifies CreateTime can't be updated.

Reply:
Also it would be a better way to show that generation doesn't need to be right.